### PR TITLE
TINKERPOP-1742 RangeByIsCountStrategy fails for ConnectiveSteps

### PR DESCRIPTION
**Pull request by dkuppitz**
https://issues.apache.org/jira/browse/TINKERPOP-1742

Fixed a bug in `RangeByIsCountStrategy`.

### CODE REVIEW

**Comment by dkuppitz** [comment]
`docker/build.sh -t -i` succeeded.

VOTE: +1

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -143,6 +143,7 @@ TinkerPop 3.2.6 (Release Date: NOT OFFICIALLY RELEASED YET)
 
 This release also includes changes from <<release-3-1-8, 3.1.8>>.
 
+* Fixed a bug in `RangeByIsCountStrategy`.
```
Comment (okram) [inline CHANGELOG.asciidoc]: Can you be a bit more explicit? -- e.g. "Fixed a bug regarding `ConnectiveSteps` in `RangeByIsCountStrategy`.

File: gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/CountStrategyTest.java [hunk 2]
```diff
@@ -95,6 +95,7 @@
                 {__.filter(__.bothE().count().is(gte(1))), __.filter(__.bothE())},
                 {__.filter(__.bothE().count().is(gt(1))), __.filter(__.bothE().limit(2).count().is(gt(1)))},
                 {__.filter(__.bothE().count().is(gte(2))), __.filter(__.bothE().limit(2).count().is(gte(2)))},
+                {__.and(__.out().count().is(0), __.in().count().is(0)), __.and(__.not(__.out()), __.not(__.in()))}
         });
```
Comment (okram) [inline gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/CountStrategyTest.java]: Perhaps one more test using both `and()` and `or()` and another pattern such as `is(gt(1))`?

**Comment by okram** [comment]
VOTE +1

**Comment by dkuppitz** [comment]
Done.

**Comment by robertdale** [comment]
VOTE +1

**Comment by spmallette** [comment]
All tests pass with `docker/build.sh -t -n -i`

VOTE +1

**Pull request closed without merging**